Add support for v4l2 on Linux so it can compile against modern
kernel headers. #363201

Patch by Kevin McCarthy <signals@gentoo.org>

--- configure.ac
+++ configure.ac
@@ -382,6 +382,10 @@
 	AC_CHECK_HEADERS(sys/videodev2.h,FOUND_OS=solaris,FOUND_OS=linux)
 fi
 
+if test "$FOUND_OS" = "linux"; then
+    AC_CHECK_HEADERS(linux/videodev2.h)
+fi
+
 #---------------------------------------------------------------------------------------------
 
 dnl ---------------------------------------------------------------------
--- utils/linux/capture/capture.h
+++ utils/linux/capture/capture.h
@@ -33,7 +33,11 @@
 #ifdef HAVE_SYS_VIDEODEV2_H
 #   include <sys/videodev2.h>
 #else
-#   include <linux/videodev.h>
+	#ifdef HAVE_LINUX_VIDEODEV2_H
+		#include <linux/videodev2.h>
+	#else
+		#include <linux/videodev.h>
+	#endif
 #endif
 
 #include "grab-ng.h"
